Ruby/Week 6/OOPsie Project/tests.rb

```diff
@@ -0,0 +1,156 @@
+    assert_equal('shiny', mermaid.favorite_style)
+    assert_equal(fluency, mermaid.fluency)
+  end
+
+  def test_dwarf_class_initialize_objects_equal
+    dwarf1 = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+    dwarf2 = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+
+    assert_equal(dwarf2, dwarf1)
+  end
+
+  def test_yeti_class_initialize_objects_not_equal
+    yeti1 = Yeti.new('Yeto', { apple: 5, hat: 1, snowboard: 3 }, "Let's go!", { snowboarding: 'expert', sledding: 'expert', hockey: 'intermediate' })
+    yeti2 = Yeti.new('Yeto', { apple: 5, hat: 1, snowboard: 3 })
+
+    refute_equal(yeti2, yeti1)
+  end
+
+  def test_give_item_success
+    mermaid = Merpeople.new('Ariel', { fork: 2, plate: 3, seashells: 5, apple: 4 })
+    dwarf = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+
+    mermaid.give_item(dwarf, "apple", 2)
+    assert_equal({ fork: 2, plate: 3, seashells: 5, apple: 2 }, mermaid.items)
+    assert_equal({ apple: 4, beer: 4, sword: 1 }, dwarf.items)
+  end
+
+  def test_give_item_fail_receiver_does_not_own_item
+    mermaid = Merpeople.new('Ariel', { fork: 2, plate: 3, seashells: 5, apple: 4 })
+    dwarf = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+    expected = "Error: Gimli isn't currently carrying forks, and as a result, can't give it to Ariel.\n"
+
+    assert_output(expected) { mermaid.give_item(dwarf, "fork", 2) }
+  end
+
+  def test_give_item_fail_giver_does_not_own_item
+    mermaid = Merpeople.new('Ariel', { fork: 2, plate: 3, seashells: 5, apple: 4 })
+    dwarf = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+    expected = "Error: Ariel isn't currently carrying swords, and as a result, can't give it to Gimli.\n"
+
+    assert_output(expected) { mermaid.give_item(dwarf, 'sword', 2) }
+  end
+
+  def test_receive_item_fail_receiver_does_not_own_item
+    mermaid = Merpeople.new('Ariel', { fork: 2, plate: 3, seashells: 5, apple: 4 })
+    dwarf = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+    expected = "Error: Gimli isn't currently carrying forks, and as a result, can't give it to Ariel.\n"
+
+    assert_output(expected) { dwarf.receive_item(mermaid, 'fork', 2) }
+  end
+
+  def test_receive_item_fail_giver_does_not_own_item
+    mermaid = Merpeople.new('Ariel', { fork: 2, plate: 3, seashells: 5, apple: 4 })
+    dwarf = Dwarf.new('Gimli', { apple: 2, beer: 4, sword: 1 })
+    expected = "Error: Ariel isn't currently carrying swords, and as a result, can't give it to Gimli.\n"
+
+    assert_output(expected) { dwarf.receive_item(mermaid, 'sword', 2) }
+  end
+
+  def test_speak
+    mermaid = Merpeople.new('Ariel', @items)
+    expected = "Ariel is speaking\n"
+    assert_output(expected) { mermaid.speak }
+  end
+
+  def test_move
+    dwarf = Dwarf.new('Gimli', @items)
+    expected = "Gimli is walking around\n"
+
+    assert_output(expected) { dwarf.move }
+  end
+
+  def test_mine
+    dwarf = Dwarf.new('Gimli', @items)
+    dwarf.mine
```

Comment:
Can use a stub/mock here, since we're technically testing the `jewels_found` hash isn't empty after `mine` is run. Currently, `mine` and `jewels_found` is tightly coupled - not great if changes were made in `mine` that would affect the returned hash.